lnwallet/chainfee/estimator.go

```diff
@@ -238,6 +238,7 @@ func (b *BtcdEstimator) Stop() error {
 	b.filterManager.Stop()
 
 	b.btcdConn.Shutdown()
+	b.btcdConn.WaitForShutdown()
```

Comment:
Hmm, I wonder if the `WaitForShutdown()` should actually be called within `Shutdown()`, in the base implementation? Or maybe there's cases where we don't want that, and that's why it's part of the interface?
Non-blocking, probably not worth spending a lot of time on, if this approach works too.

Reply:
yeah also find it weird - think it's a pattern used in `btcwallet` almost a decade ago, not sure about its intention but I do agree we should change it in the base layer as it's prone to errors, sth to look at when implementing TRUC.